### PR TITLE
Nicknames proportion

### PR DESCRIPTION
**Pull request by albrja**
### Adds function to calculate nickname proportion constant to be used in pseudopeople for noise scaling.
- *Category*: Other
- *JIRA issue*: [MIC-3861](https://jira.ihme.washington.edu/browse/MIC-3861)
- *Research reference*: N/A

### Changes and notes
-Adds nicknames csv
-Adds function to calculate proportion of names that have a nickname

### Verification and Testing
N/A

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+def get_nicknames_proportion():
+    # This function calculates the proportion of names that have potential nicknames. This will return a float that
+    # that will be used as a constant to scale the nicknames noise function in the pseudopeople packages.
+
+    art = Artifact(paths.DEFAULT_ARTIFACT)
+    names = art.load(data_keys.SYNTHETIC_DATA.FIRST_NAMES).reset_index()["name"]
+    nicknames = _load_nicknames_data()
+
+    proportion_of_nicknames = len(nicknames.index) / len(names.unique())
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:24]: You don't want unique names in the denominator. You want all names.

**Review by rmudambi — CHANGES_REQUESTED** [review]

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    # that will be used as a constant to scale the nicknames noise function in the pseudopeople packages.
+
+    art = Artifact(paths.DEFAULT_ARTIFACT)
+    names = art.load(data_keys.SYNTHETIC_DATA.FIRST_NAMES).reset_index()["name"]
+    # Load and format nicknames dataset
+    names_with_nicknames = pd.read_csv(paths.NICKNAMES_DATA_PATH)
+    names_with_nicknames = names_with_nicknames["name"].apply(lambda x: x.title())
+
+    # Proportion
+    nickname_eligible = names.isin(names_with_nicknames)
```
Comment (NathanielBlairStahn) [inline src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:18]: Do you know that all the names in the artifact are title case?

Comment (albrja) [reply src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:18]: Yes

Comment (rmudambi) [reply src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:18]: Are you sure? There are a lot of African American names like DeAngelo, LaToya etc. that aren't title case. I suspect those names aren't in our nicknames csv though, so it probably doesn't matter.

File: src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+from vivarium_census_prl_synth_pop.constants import data_keys, paths
+
+
+def get_nicknames_proportion():
+    # This function calculates the proportion of names that have potential nicknames. This will return a float that
+    # that will be used as a constant to scale the nicknames noise function in the pseudopeople packages.
+
+    art = Artifact(paths.DEFAULT_ARTIFACT)
+    names = art.load(data_keys.SYNTHETIC_DATA.FIRST_NAMES).reset_index()["name"]
```
Comment (zmbc) [inline src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:12]: This ignores the "freq" column which is what makes some names more common than others. I think we want to correct this.

File: src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py (hunk 2 above)
Comment (NathanielBlairStahn) [reply src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:18]: Yeah, I was assuming we would do a case-insensitive match to determine whether a name is eligible for a nickname, e.g. convert all the simulation names to lower case to match the nicknames CSV. And yeah, converting all the chosen nicknames to title case seems reasonable.

File: src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
 
-    art = Artifact(paths.DEFAULT_ARTIFACT)
-    names = art.load(data_keys.SYNTHETIC_DATA.FIRST_NAMES).reset_index()["name"]
+    data_files = []
+    for file in data_dir.glob("*.parquet"):
+        data_files.append(file)
+    names = []
+    for file in data_files:
+        df = pd.read_parquet(file)
+        df = df.loc[df["year"] == 2030, "first_name"]
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/data/nicknames_proportion.py:20]: Don't worry about changing this, but if we re-run this, we should use the strategy to load only the 2030 year from the file that we are already using to load in our interface functions.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**